opencart 1.5.6.4/upload/admin/controller/module/smaily_for_opencart.php

```diff
@@ -122,7 +183,36 @@ public function index() {
         );
         $this->response->setOutput($this->render());
     }
-
+    protected function handleCustomerSync() {
+        if (!$this->user->hasPermission('modify', 'module/smaily_for_opencart')) {
+            return;
+        }
+        // Load Smaily admin model for saving settings.
+        $this->load->model('smailyforopencart/admin');
+        // Declare customer sync objects.
+        $customer_sync_enabled = $this->request->post['smaily_for_opencart_sync_enabled'];
+        // For if no customer sync fields are selected.
+        $customer_sync_fields = array();
+        if (!empty($this->request->post['smaily_for_opencart_sync_fields']) && is_array($this->request->post['smaily_for_opencart_sync_fields'])) {
+            $customer_sync_fields = $this->request->post['smaily_for_opencart_sync_fields'];
+        }
+        $customer_sync_token = $this->request->post['smaily_for_opencart_sync_token'];
+        // Data validation.
+        $customer_sync_enabled = ($customer_sync_enabled == 0 || $customer_sync_enabled == 1) ? $customer_sync_enabled : 0;
+        $customer_sync_fields = array_intersect(['firstname', 'lastname', 'telephone', 'date_added'], $customer_sync_fields);
+        // Cron token field validation and sanitization.
+        $customer_sync_token = !empty($customer_sync_token) ? $this->db->escape($customer_sync_token) : uniqid();
+        // Add declared objects to array.
+        $settings = [
+          'sync_enabled' => $customer_sync_enabled,
```

Comment:
Having `sync_` before every setting does not provide any value. To keep consistency with API credentials saving, where you do not use prefixes, I suggest removing `sync_` prefix from each setting.